nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -1933,11 +1934,16 @@ private Description matchDereference(
       return Description.NO_MATCH;
     }
     if (mayBeNullExpr(state, baseExpression)) {
-      String message = "dereferenced expression " + baseExpression.toString() + " is @Nullable";
+      final String message =
+          "dereferenced expression " + baseExpression.toString() + " is @Nullable";
+      ErrorMessage errorMessage = new ErrorMessage(MessageTypes.DEREFERENCE_NULLABLE, message);
+
+      handler.onPrepareErrorMessage(baseExpression, state, errorMessage);
+
       return createErrorDescriptionForNullAssignment(
-          MessageTypes.DEREFERENCE_NULLABLE,
+          errorMessage.messageType,
           derefExpression,
-          message,
+          errorMessage.message,
```

Comment:
We should probably change all the `createErrorDescription[...]` methods to just take `ErrorMessage` rather than the separate components. But let's leave that for a follow up PR.

Reply:
Yes, definitely.